lib/climate_control.rb

```diff
@@ -1,16 +1,44 @@
-require "climate_control/environment"
 require "climate_control/errors"
-require "climate_control/modifier"
 require "climate_control/version"
+require "monitor"
 
 module ClimateControl
-  @@env = ClimateControl::Environment.new
+  extend self
 
-  def self.modify(environment_overrides = {}, &block)
-    Modifier.new(env, environment_overrides, &block).process
+  SEMAPHORE = Monitor.new
+  private_constant :SEMAPHORE
+
+  def modify(environment_overrides = {}, &block)
+    environment_overrides = environment_overrides.transform_keys(&:to_s)
+
+    SEMAPHORE.synchronize do
+      previous = ENV.to_hash
+      middle = {}
+      copy environment_overrides
+      middle = ENV.to_hash
+      block.call
+    ensure
+      after = ENV
```

Comment:
Yeah, I like this, giving it a new name makes it easier to understand how it relates to the other variables.

Reply:
Thanks